spec.emu

```diff
@@ -1,23 +1,336 @@
+                1. If _part_ has a [[Unit]] field, then
+                  1. Set _concat_ to the string-concatenation of _concat_ and _part_.[[Unit]].
+              1. Append the new Record { [[Type]]: _field_, [[Value]]: _concat_ } to the end of _result_.
+          1. Set _result_ to ! CreatePartsFromList(_durationFormat_.[[ListFormat]], _result_).
+        1. Return _result_.
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-intl-durationformat-constructor">
+    <h1>The Intl.DurationFormat Constructor</h1>
+
+    <p>The DurationFormat constructor is the <dfn>%DurationFormat%</dfn> intrinsic object and a standard built-in property of the Intl object. Behaviour common to all service constructor properties of the Intl object is specified in <emu-xref href="#sec-internal-slots"></emu-xref>.</p>
+
+    <emu-clause id="sec-Intl.DurationFormat">
+      <h1>Intl.DurationFormat ( [ _locales_ [, _options_ ] ] )</h1>
+
+      <p>When the `Intl.DurationFormat` function is called with optional arguments _locales_ and _options_, the following steps are taken:</p>
+
+      <emu-alg>
+        1. If NewTarget is *undefined*, throw a *TypeError* exception.
+        1. Let _durationFormat_ be ? OrdinaryCreateFromConstructor(NewTarget, `"%DurationFormatPrototype%"`, &laquo; [[InitializedDurationFormat]], [[Locale]], [[NumberingSystem]], [[Style]], [[Fields]], [[Templates]] &raquo;).
+        1. Let _requestedLocales_ be ? CanonicalizeLocaleList(_locales_).
+        1. If _options_ is *undefined*, then
+          1. Let _options_ be ObjectCreate(*null*).
+        1. Else
+          1. Let _options_ be ? ToObject(_options_).
+        1. Let _opt_ be a new Record.
+        1. Let _matcher_ be ? GetOption(_options_, `"localeMatcher"`, `"string"`, &laquo; `"lookup"`, `"best fit"` &raquo;, `"best fit"`).
+        1. Set _opt_.[[localeMatcher]] to _matcher_.
+        1. Let _numberingSystem_ be ? GetOption(_options_, `"numberingSystem"`, `"string"`, `undefined`, `undefined`).
+          1. If _numberingSystem_ does not match the Unicode Locale Identifier `type` nonterminal, throw a `RangeError` exception.
+        1. Set _opt_.[[nu]] to _numberingSystem_.
+        1. Let _localeData_ be %DurationFormat%.[[LocaleData]].
+        1. Let _r_ be ResolveLocale(%DurationFormat%.[[AvailableLocales]], _requestedLocales_, _opt_, %DurationFormat%.[[RelevantExtensionKeys]], _localeData_).
+        1. Let _locale_ be r.[[locale]].
+        1. Set _durationFormat_.[[Locale]] to _locale_.
+        1. Set _durationFormat_.[[NumberingSystem]] to _r_.[[nu]].
+        1. Let _style_ be ? GetOption(_options_, `"style"`, `"string"`, &laquo; `"long"`, `"short"`, `"narrow"`, `"dotted"` &raquo;, `"long"`).
+        1. Set _durationFormat_.[[Style]] to _style_.
+        1. If _style_ is `"dotted"`, then
+          1. Set _durationFormat_.[[Templates]] to a new empty Record.
```

Comment:
I guess this is a TODO, to fill in with the actual logic?

Reply:
This isn't a TODO, it's an oopsie. Idk why I didn't put the actual logic in there.

Reply:
Ah! Guess what? This condition is now unnecessary. Perfect.